src/lib.rs

```diff
@@ -221,6 +249,20 @@ fn basic() {
 }
 
 #[test]
+fn macros() {
+    fn a() -> Either<u32, u32> {
+        let x = left!(Right::<u32, u32>(1337));
```

Comment:
How would that work without type hints? What is `Right(1337)` supposed to be? You do need type hints in these situations.